### ISSUE TITLE
Configurable Spacings

### ISSUE THREAD

**Issue by detzt** [DotNetAnalyzers/StyleCopAnalyzers#2233]
It would be great to configure spacing options like SA1000 in the stylecop.json file.
Currently there is a "spacingRules" entry in the schema, but no settings are available.

**Comment by bjornhellander** [comment]
A couple of general comments:
- Configurability increases development and maintainance work in this project.
- The details of the spacing style are rarely important, as long as your IDE does not fight against your desired style.
- I think that StyleCop.Analyzers already supports the de facto code style for C#.

Having said that, what additional style would you like to have support for? Any specific reason for that?

**Comment by detzt** [comment]
Maybe it is because we are used to other languages, but our team prefers to have no space between if/for/while... and the round opening parentheses, like `if(condition) {...`.

I understand your point, but I thought that the "spacingRules" section in the schema has exactly the purpose to adjust these kind of things ;)

**Comment by bjornhellander** [comment]
You are of course absolutely correct. The only question is if it's worth the extra work.
For me, this is not important enough to warrant configurability, but lets see if anyone else has an opinion!

**Comment by sharwell** [comment]
I don't think we should take on this issue in StyleCop Analyzers since it looks like the next version of Roslyn already plans to support it:

* https://docs.microsoft.com/en-us/visualstudio/ide/create-portable-custom-editor-options
* dotnet/roslyn#15020
* dotnet/roslyn#15029
